apps/studio/components/interfaces/DiskManagement/DiskManagementForm.tsx

```diff
@@ -0,0 +1,442 @@
+                />
+                <NoticeBar
+                  type="default"
+                  visible={noPermissions}
+                  title="You do not have permission to update disk configuration"
+                  description="Please contact your organization administrator to update your disk configuration"
+                />
+              </div>
+            ) : null}
+            {/* TESTING */}
+            {/* <div className="bg-alternative border p-3 rounded text-xs font-mono">
+              <div>project.infra_compute_size: {project?.infra_compute_size}</div>
+              <div>project.status: {project?.status}</div>
+              <div className="mt-2">form status:</div>
+              <pre className="mt-2 text-xs">{JSON.stringify(form.getValues(), null, 2)}</pre>
+            </div> */}
+            {/* TESTING */}
+            <Separator />
+            <ComputeSizeField form={form} disabled={disableComputeInputs} />
+            <Separator />
+            <DiskCountdownRadial />
+            <SpendCapDisabledSection />
+            <DiskSizeField
+              form={form}
+              disableInput={disableDiskInputs}
+              setAdvancedSettingsOpenState={setAdvancedSettingsOpenState}
+            />
+            <Separator />
+            <Collapsible_Shadcn_
+              // TO DO: wrap component into pattern
+              className="-space-y-px"
+              open={advancedSettingsOpen}
+              onOpenChange={() => setAdvancedSettingsOpenState((prev) => !prev)}
+            >
+              <CollapsibleTrigger_Shadcn_ className="px-8 py-3 w-full border flex items-center gap-6 rounded-t data-[state=closed]:rounded-b group justify-between">
+                <div className="flex flex-col items-start">
+                  <span className="text-sm text-foreground">Advanced disk settings</span>
+                  <span className="text-sm text-foreground-light">
+                    Specify additional settings for your disk, including IOPS, throughput, and disk
+                    type.
+                  </span>
+                </div>
+                <ChevronRight
+                  size={16}
+                  className="text-foreground-light transition-all group-data-[state=open]:rotate-90"
+                  strokeWidth={1}
+                />
+              </CollapsibleTrigger_Shadcn_>
+              <CollapsibleContent_Shadcn_ className="data-[state=open]:border flex flex-col gap-8 px-8 py-8 transition-all data-[state=closed]:animate-collapsible-up data-[state=open]:animate-collapsible-down">
+                <NoticeBar
+                  type="default"
+                  visible={
+                    RESTRICTED_COMPUTE_FOR_THROUGHPUT_ON_GP3.includes(form.watch('computeSize')) &&
+                    subscription?.plan.id !== 'free'
+                  }
+                  title="Advanced Disk settings require LARGE Compute size or above"
```

Comment:
Also - i seem to be able to change the storage type while i'm _not_ on a large compute - is this intended?
the iops and throughput input fields are disabled correctly

Reply:
I'll check with team. I brought this up, nobody confirmed yet